firmware/library/L1_Drivers/dac.hpp

```diff
@@ -1,19 +1,76 @@
+// Dac is a driver for the Digital-To-Analog converter for the LPC40xx chip
 #pragma once
-
-#include <cstdint>
-#include <cstdio>
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/pin_configure.hpp"
 
 class DacInterface
 {
  public:
-    virtual bool Initialize();
-    virtual bool WriteDac(uint16_t dac_output);
-    virtual bool SetVoltage(float voltage);
-    // Sine Demonstration Function
-    virtual void SineDac(uint32_t frequency_hz);
-    // Sawtooth Demonstration Function
-    virtual void SawtoothDac(uint32_t frequency_hz);
-    // Triangle Demonstration Function
-    virtual void TriangleDac(uint32_t frequency_hz);
-    virtual void SetBias(bool bias_level);
+    enum BiasLevel: uint8_t
+    {
+        kBiasHigh  = 0,
+        kBiasLow   = 1
+    };
```

Comment:
I still think you should move this to the Dac class, since it is specific to the LPC4078. Also, make this into a `enum class`.

Reply:
I use it as a parameter to pass into SetBias(BiasLevel bias_level); 
Thus I think I need it to be in the interface.

Reply:
Ah, I see. Then remove SetBias from the interface then. It should only be in the implementation of the driver.